array/DNAm/analysis/EWAS/testCellComposition.r

```diff
@@ -0,0 +1,165 @@
+##---------------------------------------------------------------------#
+##
+## Title: Test cell type composition against disease status
+##
+## Purpose of script: calculate cellular composition of brain cell types
+## and test for differences between cases and controls
+##
+##---------------------------------------------------------------------#
+
+#----------------------------------------------------------------------#
+# DEFINE CETYGO FUNCTION FOR BRAIN TISSUE
+#----------------------------------------------------------------------#
+
+calcBrainComposition <- function(betas){
+  
+  predPropAll<-list() # store output in list
+
+  # run CETYGO using both ANOVA and IDOL methods and all available cell ref panels
+    for(method in names(modelBrainCoef)){
+        for(j in 1:length(modelBrainCoef[[method]])){
+        if(!is.null(modelBrainCoef[[method]][[j]])){
+                predPropAll[[method]][[j]]<-projectCellTypeWithError(betas, modelBrainCoef[[method]][[j]])
+        }
+        } 
+    }
+    predPropBest<-cbind(predPropAll[["ANOVA"]][[1]][,c("NeuNNeg_SOX10Neg", "NeuNPos")], predPropAll[["IDOL"]][[5]][,c("NeuNNeg_Sox10Neg_IRF8Pos","NeuNNeg_Sox10Neg_IRF8Neg")], predPropAll[["ANOVA"]][[3]][,c("SATB2Neg", "SATB2Pos")], predPropAll[["ANOVA"]][[6]][,c("NeuNNeg", "NeuNPos_SOX6Pos", "NeuNPos_SOX6Neg")], predPropAll[["IDOL"]][[4]][,"NeuNNeg_SOX10Pos"])
+    colnames(predPropBest)[10]<-"NeuNNeg_SOX10Pos"
+
+    # add in cetygo scores
+    predPropBest<-as.data.frame(predPropBest)
+    predPropBest$CETYGO<-predPropAll[["ANOVA"]][[1]][,c("CETYGO")]
+    return(predPropBest)
+}
+
+
+#----------------------------------------------------------------------#
+# LOAD PACKAGES
+#----------------------------------------------------------------------#
+
+library(CETYGO)
+library(gridExtra)

```

Comment:
Is this library actually being used? I can't see any usage of `gridExtra` anywhere in the plotting section